pkg/tsdb/mssql/mssql.go

```diff
@@ -146,5 +148,14 @@ func (t *mssqlQueryResultTransformer) TransformQueryResult(columnTypes []*sql.Co
 }
 
 func (t *mssqlQueryResultTransformer) TransformQueryError(err error) error {
+	// go-mssql overrides source error why we currently match on string
+	// ref https://github.com/denisenkom/go-mssqldb/blob/045585d74f9069afe2e115b6235eb043c8047043/tds.go#L904
+	if strings.HasPrefix(strings.ToLower(err.Error()), "unable to open tcp connection with host") {
+		t.log.Error("query error", "err", err)
+		return errConnectionFailed
+	}
+
 	return err
 }
+
+var errConnectionFailed = errors.New("failed to connect - please inspect Grafana server log for details")
```

Comment:
The same error is also defined in tsdb/sqleng, should we extract into _one_ definition?

Reply:
Good point